src/cascadia/TerminalApp/AppCommandlineArgs.cpp

```diff
@@ -217,11 +217,22 @@ void AppCommandlineArgs::_buildSplitPaneParser()
         // _getNewTerminalArgs might reset them while parsing a commandline
         if ((*_horizontalOption || *_verticalOption) && (_splitHorizontal))
         {
-            args->SplitStyle(SplitState::Horizontal);
+            if (_splitHorizontal)
+            {
+                args->SplitStyle(SplitState::Horizontal);
+            }
+            else if (_splitVertical)
+            {
+                args->SplitStyle(SplitState::Horizontal);
+            }
+            else
+            {
+                args->SplitStyle(SplitState::Automatic);
+            }
         }
         else
         {
-            args->SplitStyle(SplitState::Vertical);
+            args->SplitStyle(SplitState::Automatic);

```

Comment:
It seems like you could do away with both this on 235 and the one on 230 by setting it to Auto _by default_ on 217, then only overriding it if `_splitH` or `_splitV`. That'll remove a bunch of empty elses, too